mona_sdk/client.py

```diff
@@ -250,8 +259,26 @@ def __init__(
             override_host=override_app_server_host
         )
         self.filter_none_fields_on_export = filter_none_fields_on_export
-        self._default_sampling_rate = default_sampling_rate
-        self._context_class_to_sampling_rate = context_class_to_sampling_rate or {}
+
+        # Data sampling.
+
+        self._sampling_config_name = sampling_config_name
+
+        sampling_config_list = (
```

Comment:
nit: I prefer to wrap these 2 lines in 

if self._sampling_config_name:

instead of relying on get_sampling_factors to return an empty list, which might happen due to a downstream bug trying to get a named config... Then we always try to get the sampling_config from the get_sampling_factors result and if there is a problem with the config we can fail (we expect when the name is used to have some config there, right?)

Reply:
The first time we initiate the client with a new config name, the config will be empty, only after calling create_sampling_factor at least once we have the config name saved to our index

Reply:
I changed like you said just had to consider that we won't necessarily get back a config doc

Reply:
not sure I follow - 
I expect create_sampling_factor to be run by a different instance of the client in a different environment, by some support software that we will automate one day to update the sampling size automatically after seeing the sizes of the data. Until then we rely on the GUI to add these configs and the implementation of the API's create rule will be just for the sake of completeness... When we run this sdk using a certain config name this is by the sdk that actually sends the data, I expect it to be preconfigured with an existing named config and not create it in the same instance.. Make sense?